storage/innobase/buf/buf0flu.cc

```diff
@@ -1300,9 +1300,21 @@ static void buf_flush_LRU_list_batch(ulint max, bool evict,
   static_assert(FIL_NULL > SRV_TMP_SPACE_ID, "consistency");
   static_assert(FIL_NULL > SRV_SPACE_ID_UPPER_BOUND, "consistency");
 
+  /* BUF_LRU_MIN_LEN (256) is too high value for low buffer pool(BP) size. For
+  example, for BP size lower than 80M and 16 K page size, the limit is more than
+  5% of total BP and for lowest BP 5M, it is 80% of the BP. Non-data objects
+  like explicit locks could occupy part of the BP pool reducing the pages
+  available for LRU. If LRU reaches minimum limit and if no free pages are
+  available, server would hang with page cleaner not able to free any more
+  pages. To avoid such hang, we adjust the LRU limit lower than the limit for
+  data objects as checked in buf_LRU_check_size_of_non_data_objects() i.e. one
+  page less than 5% of BP. */
+  auto pool_limit = buf_pool.curr_size / 20 - 1;
+  auto buf_lru_min_len = std::min(pool_limit, (ulint)BUF_LRU_MIN_LEN);
```

Comment:
I think that we should use `size_t` instead of `auto` here, and write `std::min<size_t>` to avoid the  cast on the constant.

Monty wants to avoid a space before assignment operators:
```c++
size_t pool_limit= buf_pool.curr_size / 20 - 1;
```

Reply:
ok, sure.